packages/ember-engines/addon/components/link-to-external.js

```diff
@@ -1,53 +1,941 @@
-import { LinkTo as RoutingLinkComponent } from '@ember/routing';
+/* eslint-disable ember/no-classic-classes */
+/* eslint-disable ember/require-tagless-components */
+/* eslint-disable ember/no-private-routing-service */
+/* eslint-disable ember/require-computed-property-dependencies */
+/* eslint-disable ember/no-component-lifecycle-hooks */
+/* eslint-disable ember/require-super-in-lifecycle-hooks */
+/* eslint-disable ember/no-assignment-of-untracked-properties-used-in-tracking-contexts */
+/* eslint-disable ember/no-classic-components */
+import { alias } from '@ember/object/computed';
+import { computed } from '@ember/object';
 import { getOwner } from '@ember/application';
-import { set, get } from '@ember/object';
-import { macroCondition, dependencySatisfies, importSync } from '@embroider/macros';
+import EmberComponent from '@ember/component';
+import { assert, deprecate, runInDebug, warn } from '@ember/debug';
+import { getEngineParent } from '@ember/engine';
+import { inject as injectService } from '@ember/service';
+import { DEBUG } from '@glimmer/env';
 
-let LinkToExternal;
-let LinkComponent;
+function intern(str) {
+  let obj = {};
+  //@ts-ignore
+  obj[str] = 1;
+  for (let key in obj) {
+      if (key === str) {
+          return key;
+      }
+  }
+  return str;
+}
+const GUID_KEY = intern(`__ember${Date.now()}`);
+// `enumerableSymbol` copied from https://github.com/emberjs/ember.js/blob/master/packages/@ember/-internals/utils/lib/symbol.ts
+// for not exported code these legacy components are dependant on.
+// Some legacy symbols still need to be enumerable for a variety of reasons.
+// This code exists for that, and as a fallback in IE11. In general, prefer
+// `symbol` below when creating a new symbol.
+function enumerableSymbol(debugName) {
+  let id = GUID_KEY + Math.floor(Math.random() * Date.now());
+  let symbol = intern(`__${debugName}${id}__`);
+  return symbol;
+}
+const HAS_BLOCK = enumerableSymbol('HAS_BLOCK');
 
-if (macroCondition(dependencySatisfies('@ember/legacy-built-in-components', '*'))) {
-  let { LinkComponent: LegacyLinkComponent } = importSync('@ember/legacy-built-in-components');
-  LinkComponent = LegacyLinkComponent;
-} else {
-  LinkComponent = RoutingLinkComponent;
+
+function isSimpleClick(event) {
+  let modifier = event.shiftKey || event.metaKey || event.altKey || event.ctrlKey;
+  let secondaryClick = event.which > 1; // IE9 may return undefined
+  return !modifier && !secondaryClick;
 }
 
-if (macroCondition(dependencySatisfies('ember-source', '>=3.24.1 || >=4.x'))) {
-  LinkToExternal = class LinkToExternal extends LinkComponent {
-    _namespaceRoute(targetRouteName) {
-      const owner = getOwner(this);
-     
-      if (!owner.mountPoint) {
-        return super._namespaceRoute(...arguments);
-      }
-      
-      const externalRoute = owner._getExternalRoute(targetRouteName);
+/**
```

Comment:
@void-mAlex is it a copy/paste of original LinkTo?

Reply:
short answer is yes, long one is it's not byte for byte for reasons of deprecation removals